torch/futures/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""
+The ``torch.futures`` package contains a ``Future`` type and corresponding
+utility functions.
+"""
+import torch
+
+
+class Future(torch._C.Future):
+    r"""
+    Wrapper around a ``torch._C.Future``.
+    """
+    def __new__(cls):
+        return super(Future, cls).__new__(cls)
+
+    def wait(self):
+        r"""
+        Block until the value of this ``Future`` is ready.
+
+        Return:
+            The value held by this ``Future``.
```

Comment:
Should we also mention this could also throw exceptions?

Reply:
Good point, will add that